wsmux/stream.go

```diff
@@ -0,0 +1,220 @@
+package wsmux
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"sync"
+	"time"
+)
+
+// DefaultCapacity is the maximum length the read buffer will accept
+const DefaultCapacity = 1024
+
+var (
+	// errWriteTimeout is returned if no ack frame arrives before the duration.
+	errWriteTimeout = fmt.Errorf("write operation timed out")
```

Comment:
I suspect people handling error might want to handle these? if they are not strictly internal and potentially interesting to consumers... let's put them in an `errors.go` file and export them...